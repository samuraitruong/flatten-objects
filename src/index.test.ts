import { flattenObject } from './index';
describe('flattenObject', () => {
  it('should return same object if there no array property', () => {
    const a = { a: 123 }
    expect(flattenObject(a)).toEqual([a])
  })

  it('should return object with empty value when array is empty', () => {
    const a = { a: 123, b: [] }
    expect(flattenObject(a, '')).toEqual([{ a: 123, b: '' }])
  })

  it('should return 2  objects if there no array property', () => {
    const a = { a: 123, t: [1, 2] }
    expect(flattenObject(a)).toEqual([{ a: 123, t: 1 }, { a: 123, t: 2 }])
  })

  it('should return 6 objects if there no array property', () => {
    const a = { a: 123, t: [1, 2], v: [1, 2, 3] }
    expect(flattenObject(a).length).toEqual(6)
    expect(flattenObject(a)).toMatchSnapshot()
  })

  it('should return 6 objects if there no array property', () => {
    const a = { a: 123, t: [1, 2], v: [1, 2, 3], k: [9, 8, 7] }
    expect(flattenObject(a).length).toEqual(18)
    expect(flattenObject(a)).toMatchSnapshot()
  })
})
