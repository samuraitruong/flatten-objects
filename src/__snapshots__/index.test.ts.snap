// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flattenObject should return 6 objects if there no array property 1`] = `
Array [
  Object {
    "a": 123,
    "t": 1,
    "v": 1,
  },
  Object {
    "a": 123,
    "t": 1,
    "v": 2,
  },
  Object {
    "a": 123,
    "t": 1,
    "v": 3,
  },
  Object {
    "a": 123,
    "t": 2,
    "v": 1,
  },
  Object {
    "a": 123,
    "t": 2,
    "v": 2,
  },
  Object {
    "a": 123,
    "t": 2,
    "v": 3,
  },
]
`;

exports[`flattenObject should return 6 objects if there no array property 2`] = `
Array [
  Object {
    "a": 123,
    "k": 9,
    "t": 1,
    "v": 1,
  },
  Object {
    "a": 123,
    "k": 8,
    "t": 1,
    "v": 1,
  },
  Object {
    "a": 123,
    "k": 7,
    "t": 1,
    "v": 1,
  },
  Object {
    "a": 123,
    "k": 9,
    "t": 1,
    "v": 2,
  },
  Object {
    "a": 123,
    "k": 8,
    "t": 1,
    "v": 2,
  },
  Object {
    "a": 123,
    "k": 7,
    "t": 1,
    "v": 2,
  },
  Object {
    "a": 123,
    "k": 9,
    "t": 1,
    "v": 3,
  },
  Object {
    "a": 123,
    "k": 8,
    "t": 1,
    "v": 3,
  },
  Object {
    "a": 123,
    "k": 7,
    "t": 1,
    "v": 3,
  },
  Object {
    "a": 123,
    "k": 9,
    "t": 2,
    "v": 1,
  },
  Object {
    "a": 123,
    "k": 8,
    "t": 2,
    "v": 1,
  },
  Object {
    "a": 123,
    "k": 7,
    "t": 2,
    "v": 1,
  },
  Object {
    "a": 123,
    "k": 9,
    "t": 2,
    "v": 2,
  },
  Object {
    "a": 123,
    "k": 8,
    "t": 2,
    "v": 2,
  },
  Object {
    "a": 123,
    "k": 7,
    "t": 2,
    "v": 2,
  },
  Object {
    "a": 123,
    "k": 9,
    "t": 2,
    "v": 3,
  },
  Object {
    "a": 123,
    "k": 8,
    "t": 2,
    "v": 3,
  },
  Object {
    "a": 123,
    "k": 7,
    "t": 2,
    "v": 3,
  },
]
`;
